[tvOS] Settings Cleanup

JPKribs opened the pull request:
### Goal

This PR originally started with the goal of fixing a slight peeve I had with the settings. All buttons in Forms would clip into their boundaries when focused. See:

<img width="527" alt="Button Clipping" src="https://github.com/user-attachments/assets/65e379ef-d05d-489d-a0aa-0e82d8101643">

Following the changes here: https://github.com/jellyfin/Swiftfin/pull/1163#pullrequestreview-2217218184, I've removed the clipping. Additionally to the clipping, I have added labels to the settings items and updated it to ensure that all sections have headers.

### Settings 

I created a custom SettingsViewForm and SettingsViewFormSection to handle the Form and Section actions from a single location. The goal was to centralize the button/form configuration in single struct so they could all be updated if there are any customization that we want to apply to all sections. These settings can be seen reflected here:

<details>

<summary>Settings Main</summary>

![Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-08-03 at 17 14 57](https://github.com/user-attachments/assets/ff762432-d043-4be2-b953-6fe5ca8db181)

1. Updated the Switch User button to indicate that it will take you to another page using the Chevron button.
2. Added the Jellyfin Section category for Jellyfin Server items (First Section)

</details>

<details>

<summary>Server Details</summary>

![Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-08-03 at 17 12 38](https://github.com/user-attachments/assets/e42bdab3-74b3-4568-8f1a-21d894c4118d)

1. Created a Checkbox Button that mirrors the Chevron Button but has a variable for true/false. This removes the need to declare the button twice as checked/unchecked while also move the checkmark to more closely resemble the Chevron buttons used elsewhere in settings.

</details>

<details>

<summary>Video Player</summary>

![Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-08-03 at 17 12 54](https://github.com/user-attachments/assets/a274c7b0-96df-43dd-8361-6a8f83400fba)

1. Change Resume Offset button Offset in the Resume category. 
2. Create Section Headers to better split out content since the Header/Footer can start looking a little weird when there are footers at the end of a section but no headers for that section.

</details>

<details>

<summary>Maximum Bitrate</summary>

![Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-08-03 at 17 13 01](https://github.com/user-attachments/assets/b8a03815-9611-416a-b283-1b0da7cf9010)

</details>

<details>

<summary>Customize</summary>

![Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-08-03 at 17 13 09](https://github.com/user-attachments/assets/6bbba758-d4c1-434a-a03e-66da767bd7b9)

</details>

<details>

<summary>Experimental</summary>

![Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-08-03 at 17 13 16](https://github.com/user-attachments/assets/083b3850-cce5-44b1-bb9c-c15b065b181b)

</details>

This last section was not part of the Setting Form Changes but instead just a padding(.horizontal) to resolve this clipping.

<details>

<summary>Server Selection</summary>

![Settings](https://github.com/user-attachments/assets/87e2240e-b9cd-4aea-9613-3580ddab9cd2)

</details>

LePips reviewed (changes requested):
Thank you for taking a look at this and the localization but I apologize to say that the clipping issue could just be fixed [1] with the following in `SplitFormWindowView` and attached to other `Form`s [2] if necessary:

```swift
Form {
    contentView()
        .eraseToAnyView()
}
.scrollClipDisabled()
```

Additionally, this keeps the `Form` styling instead of moving to `Button`s in a `VStack`.

---

[1] That modifier came in tvOS 17 which I just now set as the minimum but I would have pursued a way to prevent the clipping somehow with `overlay` or whatever and still use `Form`.

[2] The entire server selection will be changed to use a `Menu` which was introduced in tvOS 17, so I wouldn't worry about my ugly replication.

JPKribs commented:
```
Form {
    contentView()
        .eraseToAnyView()
}
.scrollClipDisabled()
```
That is so much cleaner of a solution than this whole build out! I can't believe I didn't find this when I researching this. Let me re-submit this with just that single line change.

JPKribs commented:
> Thank you for taking a look at this and the localization but I apologize to say that the clipping issue could just be fixed [1] with the following in `SplitFormWindowView` and attached to other `Form`s [2] if necessary:
> 
> ```swift
> Form {
>     contentView()
>         .eraseToAnyView()
> }
> .scrollClipDisabled()
> ```
> 
> Additionally, this keeps the `Form` styling instead of moving to `Button`s in a `VStack`.
> 
> [1] That modifier came in tvOS 17 which I just now set as the minimum but I would have pursued a way to prevent the clipping somehow with `overlay` or whatever and still use `Form`.
> 
> [2] The entire server selection will be changed to use a `Menu` which was introduced in tvOS 17, so I wouldn't worry about my ugly replication.

@LePips I've moved back to just using the standard form items again with that one `.scrollClipDisabled()` change. This resolves the exact peeve I had with this so I am happy with the change. I've left in my other items of adding Labels and Section headers but let me know if you want me to drop those as well. 

Lastly, I added that padding to the Server Selection Form. Let me know if you want me to keep the padding or if that should also just be `.scrollClipDisabled()`.

I've updated the original post with screenshots to reflect the current state of this PR!

LePips commented on Swiftfin tvOS/Views/ServerDetailView.swift:
```diff
@@ -35,36 +35,49 @@ struct EditServerView: View {
-                        trailing: viewModel.server.name
-                    )
+
+                Section {
```
Please use the other init instead. I know there are many places throughout that use the other one but I've been moving to this init instead.

```swift
Section(L10n.server) {
    // ...
}
```

LePips commented on Swiftfin tvOS/Views/ServerDetailView.swift:
```diff
@@ -35,36 +35,49 @@ struct EditServerView: View {
-                    )
+
+                Section {
+                    Button {} label: {
```
We don't need just the server name to be focusable, but the user on the settings view did in order to have better scrolling to/from the tab bar.

JPKribs replied:
Leaving this as button still allows the cursor to go over it but it doesn't focus correctly. I changed this from a button to just the view and I believe this works as intended now.

LePips commented on Swiftfin tvOS/Views/SettingsView/IndicatorSettingsView.swift:
```diff
@@ -34,13 +34,15 @@ struct IndicatorSettingsView: View {
 
                 Section {
```
Since you're here, please use the other init.

LePips commented on Swiftfin tvOS/Views/SettingsView/MaximumBitrateSettingsView.swift:
```diff
@@ -24,21 +24,19 @@ struct MaximumBitrateSettingsView: View {
             }
             .contentView {
+
                 Section {
```
Since you're here, please use the other init for the header. Since the footer has a conditional, it does not need to be changed,

JPKribs replied:
For the Footers I am a little lost. Here is what I am seeing as options for Footer usage:

<img width="646" alt="Screenshot 2024-08-06 at 13 43 51" src="https://github.com/user-attachments/assets/55ec40f6-84a2-433a-99d0-ec5942510e5d">

The version I think you are looking for looks like this? But I am getting some syntax errors:

<img width="989" alt="Screenshot 2024-08-06 at 13 44 22" src="https://github.com/user-attachments/assets/6a143be5-b261-4125-9856-fd8cf262dc33">

The original version looks like this if we want to retain this. Otherwise, I am more than happy to change this to keep everything consistent!

<img width="617" alt="Screenshot 2024-08-06 at 13 44 50" src="https://github.com/user-attachments/assets/1f761b39-392d-48ea-a6b5-be3d4c418213">

LePips replied:
Ah I apologize as I must have been mixed up with the inits with footers. We can leave this as is.

LePips commented on Swiftfin tvOS/Views/SettingsView/SettingsView.swift:
```diff
@@ -51,14 +51,26 @@ struct SettingsView: View {
+                        }
                     }
+
+                } header: {
```
Since you're here, please use the other init and you can do more throughout this file if you would like to.

LePips commented on Swiftfin tvOS/Views/SettingsView/VideoPlayerSettingsView.swift:
```diff
@@ -43,46 +43,54 @@ struct VideoPlayerSettingsView: View {
             .contentView {
 
                 Section {
```
Since you're here, please use the other init for both header and footer.

LePips commented on Swiftfin tvOS/Views/SettingsView/VideoPlayerSettingsView.swift:
```diff
@@ -43,46 +43,54 @@ struct VideoPlayerSettingsView: View {
+                    L10n.resumeOffsetDescription.text
                 }
 
                 Section {
```
Since you're here, please use the other init for both header and footer.

JPKribs replied:
I believe this is the same item as https://github.com/jellyfin/Swiftfin/pull/1163#discussion_r1706031863 so I will base this change on that discussion!

LePips replied:
https://github.com/jellyfin/Swiftfin/pull/1163#discussion_r1706053161

LePips commented on Swiftfin tvOS/Components/SplitFormWindowView.swift:
```diff
@@ -31,6 +31,7 @@ struct SplitFormWindowView: View {
             .if(descriptionTopPadding) { view in
                 view.padding(.top)
             }
+            .scrollClipDisabled()
```
This does fix the issue like I said earlier but now introduces a different clipping behavior for long lists when cells clip out. The documentation comment for this modifier has a solution for custom clipping behavior so we'll have to take a look at something like that later with a desirable gradient mask.

JPKribs replied:
Sorry for my confusion. Do you want me to leave this in, take this out, or would you look me to look into custom clipping behavior? Happy to help out wherever I can!

LePips replied:
Leave this in to fix the original clipping issue. The new clipping issue was something I just noticed while testing this and we can fix later.

LePips commented on Swiftfin tvOS/Views/SettingsView/VideoPlayerSettingsView.swift:
```diff
@@ -43,46 +43,54 @@ struct VideoPlayerSettingsView: View {
-                    $0.secondLabel
+                    Toggle(L10n.pauseOnBackground, isOn: $pauseOnBackground)
+                    Toggle(L10n.playOnActive, isOn: $playOnActive)
+                } header: {
```
Please use other init.

JPKribs replied:
I believe this is the same item as https://github.com/jellyfin/Swiftfin/pull/1163#discussion_r1706031863 so I will base this change on that discussion!

LePips replied:
https://github.com/jellyfin/Swiftfin/pull/1163#discussion_r1706053161

LePips commented on Swiftfin tvOS/Components/CheckButton.swift:
```diff
@@ -0,0 +1,62 @@
+
+import SwiftUI
+
+struct CheckButton: View {
```
I don't feel like this serves much purpose in the area that it's currently being used. I would also prefer we move to pickers for select-able items, especially now that we can use `Menu` or `pickerStyle(.menu)`.

JPKribs replied:
Oh nice the Menu items look a lot better! I tried to take a crack at this. Is this kind of what you were looking for:
![Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-08-06 at 15 01 03](https://github.com/user-attachments/assets/2c2ad4d3-226d-46c7-8ffd-0b10772a0b0b)

JPKribs replied:
I've included this working version on my latest commit. It's rough around the edges with now and I don't like how the box has a larger button box around it but for the menu specific drop down I think that shows the basic of what I think you're suggesting.

JPKribs commented:
@LePips It looks like I can for the SettingView > Username that I did here: https://github.com/jellyfin/Swiftfin/pull/1163#discussion_r1705929277

The focus appears to work as we expect it to but I would agree it feels a little weird. Let me know if you want me to include this here:

https://github.com/user-attachments/assets/a823f92d-2201-4c9e-b1a3-4affc1ace8eb

LePips reviewed (approved):
Everything looks good as is. The user button should be changed to match iOS soon and I'll probably clean up some other things here as well. I'm experiencing some bugs in the iOS 18 simulator around losing focus when popping a settings view but that can be fixed later.

Thanks for doing this work and going through my requests!
